### PR TITLE
Align OTA cluster XML with the specification.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
This fixes the enum names to match.
    
The difference in the ProviderLocation name is being addressed in the spec in
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7580

REVIEW NOTE: The first changeset was created manually (though with some `git sed` involved).  The second and third are auto-generated.

### CODE REVIEW

**Review by tcarmelveilleux — DISMISSED** [review]
Do not rename all these fields in the XML and data model: this is a large backwards compatibility break.

**Comment by bzbarsky-apple** [comment]
@tcarmelveilleux Then we need to rename them in the spec.

**Review by yunhanw-google — APPROVED** [review]

**Comment by andy31415** [comment]
Applied `skip-protocol-compatibility` because these are structure renames (so field changes are actually type renames)

File: src/app/CompatEnumNames.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace OtaSoftwareUpdateProvider {
+using OTAApplyUpdateAction = ApplyUpdateActionEnum;
+using OTADownloadProtocol = DownloadProtocolEnum;
+using OTAQueryStatus = StatusEnum;
+} // namespace OtaSoftwareUpdateProvider
```
Comment (tcarmelveilleux) [inline src/app/CompatEnumNames.h:33]: Can you please add comments (written after initial push) for which PR introduced this and why?

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/CompatEnumNames.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/CompatEnumNames.h]: Sure thing.

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**